rhino/src/main/java/org/mozilla/javascript/optimizer/DefaultLinker.java

```diff
@@ -111,14 +115,26 @@ private GuardedInvocation getPropertyInvocation(
             mh =
                     bindStringParameter(
                             lookup, mType, ScriptRuntime.class, "setObjectProp", 1, op.getName());
+        } else if (op.isOperation(RhinoOperation.SETSUPER)) {
+            mh =
+                    bindStringParameter(
+                            lookup, mType, ScriptRuntime.class, "setSuperProp", 1, op.getName());
         } else if (op.isOperation(RhinoOperation.GETELEMENT)) {
             mh = lookup.findStatic(ScriptRuntime.class, "getObjectElem", mType);
+        } else if (op.isOperation(RhinoOperation.GETELEMENTSUPER)) {
+            mh = lookup.findStatic(ScriptRuntime.class, "getSuperElem", mType);
         } else if (op.isOperation(RhinoOperation.GETINDEX)) {
             mh = lookup.findStatic(ScriptRuntime.class, "getObjectIndex", mType);
+        } else if (op.isOperation(RhinoOperation.GETINDEXSUPER)) {
+            mh = lookup.findStatic(ScriptRuntime.class, "getSuperIndex", mType);
```

Comment:
Removed completely that case because it can never happen!

The reason is that functions with `super` will always use activation, and `Optimizer` will not optimize functions that do. Therefore, we will never hit a case of having both properties `ISNUMBER_PROP` and `SUPER_PROPERTY_ACCESS` at the same time.
This means that, if we have a `super` access, we will not have an extra-optimized path for integer properties and we will go to the generic "element" case, but we think it's fine. We're talking about accessing things like `super[0]` cases here, not something that should be very common.

We haven't added any assertion or check that both properties are not true at the same time because, in any case, the generic "getelem" case can handle integer/double, so even if the optimizer changes, this won't break - it might be slower than for non-super access, but it is correct.